src/pages/iou/ReceiptSelector/index.native.js

```diff
@@ -179,74 +188,60 @@ function ReceiptSelector(props) {
             })
             .then((photo) => {
                 IOU.setMoneyRequestReceipt(`file://${photo.path}`, photo.path);
-                IOU.navigateToNextPage(props.iou, iouType.current, reportID.current, props.report);
+                IOU.navigateToNextPage(props.iou, iouType, reportID, props.report);
             })
             .catch(() => {
                 showCameraAlert();
             });
-    };
+    }, [flash, iouType, props.iou, props.report, reportID, translate]);
 
     Camera.getCameraPermissionStatus().then((permissionStatus) => {
         setPermissions(permissionStatus);
     });
 
-    const cameraComponent = () => (
-        <Camera
-            ref={camera}
-            device={device}
-            style={[styles.cameraView]}
-            isActive
-            photo
-        />
-    );
-
-    const loadingComponent = () => (
-        <View style={[styles.cameraView]}>
-            <ActivityIndicator
-                size="large"
-                style={{flex: 1}}
-                color={themeColors.textSupporting}
-            />
-        </View>
-    );
-
-    const permissionComponent = () => (
-        <View style={[styles.cameraView, styles.permissionView]}>
-            <Hand
-                width={152}
-                height={200}
-                style={{paddingBottom: 20}}
-            />
-            <Text style={[styles.textReceiptUpload]}>{translate('receipt.takePhoto')}</Text>
-            <Text style={[styles.subTextReceiptUpload]}>{translate('receipt.cameraAccess')}</Text>
-            <PressableWithFeedback
-                accessibilityLabel={translate('receipt.givePermission')}
-                accessibilityRole="button"
-            >
-                <Button
-                    medium
-                    success
-                    text={translate('receipt.givePermission')}
-                    style={[styles.p9, styles.pt5]}
-                    onPress={() => {
-                        if (permissions === 'not-determined') {
-                            Camera.requestCameraPermission().then((permissionStatus) => {
-                                setPermissions(permissionStatus);
-                            });
-                        } else {
-                            Linking.openSettings();
-                        }
-                    }}
-                />
-            </PressableWithFeedback>
-        </View>
-    );
-
     const getCameraView = () => {
```

Comment:
Can we inline this?

Reply:
I think we have to pick between inline and this suggestion: https://github.com/Expensify/App/pull/23046#discussion_r1281207365